src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -240,6 +240,7 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 var antiforgeryTag = Generator.GenerateAntiforgery(ViewContext);
                 if (antiforgeryTag != null)
                 {
+                    ViewContext.FormContext.HasAntiforgeryToken = true;
```

Comment:
Not sure if you noticed but I updated an existing test with a condition https://github.com/aspnet/Mvc/pull/4626/files/39f290193310e15bc3cb050092de85da4dc378e5#diff-c3f8a68da4023116edc455c5e781fc80R95

Would it suffice?